### ISSUE TITLE
False Positive (File Access)

### ISSUE THREAD

**Issue by Schwad** [presidentbeef/brakeman#990]
- **Rails 5.0.1**
- **Ruby 2.3.3**

Howdy all,

Just here to do my due diligence as I work through my company's brakeman scanner report today. 

[This page suggested](http://brakemanscanner.org/docs/ignoring_false_positives/) that if I encounter a false positive I should report it here. 

The error I've recorded in a fair few places refers to [File Access](http://brakemanscanner.org/docs/warning_types/file_access/), specifically with the warning: `Model attribute used in file name.`

In a segment of our code we use a method that generates a "footer barcode" which is displayed at the bottom of files and in the file name. We pass elements into this method. A whitewashed example would be:

```ruby
# ../app/models/generic_model.rb

def my_generator
  my_code = MyGeneratingClass.render_barcode(environment_variable_one, environment_variable_two, "STATIC STRING", User.find(current_user_id).attribute_one, resource.association_one.attribute_two)

  path = "#{Rails.root}/tmp/#{ my_code }.pdf"
  create_attachment file: File.new(path, 'r')
end
```

I believe this is a false positive since at no point are we passing along any attributes that the use would have any control over. I understand if there's no 'fix' for this, but I really appreciate the work you guys do on this gem and wanted to contribute!

**Comment by presidentbeef** [comment]
Hi Nick,

I'm guessing it's warning about the `User.find(current_user_id).attribute_one` call? And it's a weak confidence warning?

It's possible to make the check a little less sensitive to arguments inside of other calls...might do that here.

**Comment by Schwad** [comment]
Hello @presidentbeef,

What I'll do by the morning is run it against that code a few times with one element at a time to find the exact trigger and update you on that, it's a medium warning so thankfully not `high` or worse.

**Comment by presidentbeef** [comment]
If you use the HTML output the value should be highlighted.

**Comment by Schwad** [comment]
I have a `medium` confidence warning triggering on `resource.association_one.attribute_two`

**Comment by Schwad** [comment]
Further (I am happy to file it in a separate issue but while we're on the topic)

I'm getting a weak confidence warning on:

```ruby
def download_file
    attachment = case params[:type]
                 when 'one_parameter' then resource.one_attachment
                 else resource.two_attachment
                 end
    send_file attachment.file.path, type: attachment.content_type, x_sendfile: true
  end
```

I am not too worked up over a weak confidence warning on this one and definitely see where it could be a problem if this creeped too far... And could even be a problem in a small case if one of those attachments might be one you want to severely restrict. 

Thanks for your awesome work! :thumbsup:

**Comment by presidentbeef** [comment]
Looks like the original warning is now weak confidence, and I believe the second issue was fixed with #1018.

I think that's the best we can do for now.

**Comment by Schwad** [comment]
@presidentbeef yeah that's perfect! 👍  Thank you so much!
